crates/zksync/core/src/lib.rs

```diff
@@ -183,47 +105,18 @@ pub struct EstimatedGas {
 }
 
 /// Estimates the gas parameters for the provided transaction.
-pub async fn estimate_gas<P: Provider<T, AnyNetwork>, T: Transport + Clone>(
-    tx: &WithOtherFields<TransactionRequest>,
-    factory_deps: Vec<Vec<u8>>,
+/// This will call `estimateFee` method on the rpc and set the gas parameters on the transaction.
+pub async fn estimate_gas<P: ZksyncProvider<T>, T: Transport + Clone>(
+    tx: &mut ZkTransactionRequest,
     provider: P,
-) -> Result<EstimatedGas> {
-    let to = tx
-        .to
-        .and_then(|to| match to {
-            alloy_primitives::TxKind::Create => None,
-            alloy_primitives::TxKind::Call(to) => Some(to),
-        })
-        .ok_or_eyre("`to` cannot be empty")?;
-    let chain_id = tx.chain_id.ok_or_eyre("`chain_id` cannot be empty")?;
-    let nonce = tx.nonce.ok_or_eyre("`nonce` cannot be empty")?;
-    let gas_price = if let Some(gas_price) = tx.gas_price {
```

Comment:
we had this before because without a `gas_price` sometimes the gas estimation would fail, iirc

Reply:
I tested locally and it didn't seem to matter for era-test-node `zks_estimateFee`